idaes/generic_models/unit_models/column_models/column.py

```diff
@@ -0,0 +1,1144 @@
+                            initialize=0.0,
+                            units=pyunits.J / (pyunits.s * pyunits.m),
+                            doc='Heat transfer rate in liquid phase')
+
+        # =====================================================================
+        # Add performance equations
+
+        # Inter-facial Area model ([m2/m3]):
+
+        self.area_interfacial = Var(self.flowsheet().time,
+                                    self.vapor_phase.length_domain,
+                                    initialize=0.9,
+                                    doc='Specific inter-facial area')
+
+        # ---------------------------------------------------------------------
+        # Geometry constraints
+
+        # Column area [m2]
+        @self.Constraint(doc="Column cross-sectional area")
+        def column_cross_section_area(blk):
+            return blk.area_column == (
+                CONST.pi * 0.25 * (blk.diameter_column)**2)
+
+        # Area of control volume : vapor side and liquid side
+        control_volume_area_definition = ''' column_area * phase_holdup.
+        The void fraction of the vapor phase (volumetric vapor holdup) and that
+        of the liquid phase(volumetric liquid holdup) are
+        lumped into the definition of the cross-sectional area of the
+        vapor-side and liquid-side control volume respectively. Hence, the
+        cross-sectional area of the control volume changes with time and space.
+        '''
+
+        if self.config.dynamic:
+            @self.Constraint(self.flowsheet().time,
+                             self.vapor_phase.length_domain,
+                             doc=control_volume_area_definition)
+            def vapor_side_area(bk, t, x):
+                return bk.vapor_phase.area[t, x] == (
+                    bk.area_column * bk.holdup_vap[t, x])
+
+            @self.Constraint(self.flowsheet().time,
+                             self.liquid_phase.length_domain,
+                             doc=control_volume_area_definition)
+            def liquid_side_area(bk, t, x):
+                return bk.liquid_phase.area[t, x] == (
+                    bk.area_column * bk.holdup_liq[t, x])
+        else:
+            self.vapor_phase.area.fix(value(self.area_column))
+            self.liquid_phase.area.fix(value(self.area_column))
+
+        # Pressure consistency in phases
+        @self.Constraint(self.flowsheet().time,
+                         self.liquid_phase.length_domain,
+                         doc='''Mechanical equilibruim: vapor-side pressure
+                                    equal liquid -side pressure''')
+        def mechanical_equil(bk, t, x):
+            return bk.liquid_phase.properties[t, x].pressure == \
+                    bk.vapor_phase.properties[t, x].pressure
+
+        # Length of control volume : vapor side and liquid side
+        @self.Constraint(doc="Vapor side length")
+        def vapor_side_length(blk):
```

Comment:
Add a `TODO` here saying we should add a way to define `length` at the unit model level and make `References` to that in the control volumes, and then clean up these equality constraints,